File: OC/内容层动画/WGOCCoreAnimation/WGOCCoreAnimation/CAReplicatorLayer动画/地球公转/WGEarthAniViewController.m
//
//  WGEarthAniViewController.m
//  WGOCCoreAnimation
//
//  Created by wanggang on 2019/5/29.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "WGEarthAniViewController.h"

@interface WGEarthAniViewController ()

@end

@implementation WGEarthAniViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"地球公转";
    self.view.backgroundColor = [UIColor blackColor];
    
    UIImageView *earthImgView = [[UIImageView alloc] initWithFrame:CGRectMake((SCREEN_WIDTH-50)/2+150, (SCREEN_HEIGHT-50)/2, 50, 50)];
    earthImgView.image = [UIImage imageNamed:@"earth"];
    
    UIImageView *sunImgView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 50, 50)];
    sunImgView.center = self.view.center;
    sunImgView.image = [UIImage imageNamed:@"sun"];
    
    CAReplicatorLayer *layer = [CAReplicatorLayer layer];
    [layer addSublayer:earthImgView.layer];
    [layer addSublayer:sunImgView.layer];
    layer.instanceCount = 3;
    layer.instanceDelay = 1;
    
    [self.view.layer addSublayer:layer];
    
    UIBezierPath *path = [UIBezierPath bezierPath];
    [path addArcWithCenter:self.view.center radius:150 startAngle:0 endAngle:M_PI*2 clockwise:YES];
    [path closePath];
    
    CAKeyframeAnimation *ani = [[CAKeyframeAnimation alloc] init];
    ani.keyPath = @"position";
    ani.path = path.CGPath;
    ani.duration = 10;
    ani.repeatCount = MAXFLOAT;
    
    [earthImgView.layer addAnimation:ani forKey:nil];
    
}



@end

File: Swift/内容层动画/WGCoreAnimation/WGCoreAnimation/CAReplicatorLayer动画/地球公转/WGEarthAniViewController.swift
//
//  WGEarthAniViewController.swift
//  WGCoreAnimation
//
//  Created by wanggang on 2019/5/8.
//  Copyright © 2019 wanggang. All rights reserved.
//

import UIKit

class WGEarthAniViewController: UIViewController {

    var earthImgView: UIImageView!
    var replicatorLayer:CAReplicatorLayer = CAReplicatorLayer()

    override func viewDidLoad() {
        super.viewDidLoad()

        title = "🌎公转"
        view.backgroundColor = UIColor.cyan
        
        earthImgView = UIImageView(frame: CGRect(x: (WgWith-50)/2+150, y: (WgHeight-50)/2, width: 50, height: 50))
        earthImgView.image = UIImage(named: "earth")
        
        let sunImgView = UIImageView(frame: CGRect(x: 0, y: 0, width: 50, height: 50))
        sunImgView.center = view.center
        sunImgView.image = UIImage(named: "sun")
        
        replicatorLayer.addSublayer(earthImgView.layer)
        replicatorLayer.addSublayer(sunImgView.layer)

        let bezierPath = UIBezierPath()
        bezierPath.addArc(withCenter: view.center, radius: 150, startAngle: 0, endAngle: CGFloat(Double.pi*2), clockwise: true)
        bezierPath.close()
        
        let ani = CAKeyframeAnimation()
        ani.keyPath = "position"
        ani.path = bezierPath.cgPath
        ani.duration = 10
        ani.repeatCount = MAXFLOAT
        
        replicatorLayer.instanceCount = 3
        replicatorLayer.instanceDelay = 1
        
        view.layer.addSublayer(replicatorLayer)
        earthImgView.layer.add(ani, forKey: nil)
        
    }
    

}
